### PR TITLE
Add: ModifyDevice method and change ip option

### PR DESCRIPTION
**Pull request by RaulTrombin**
json command:
```
{
  "command": "ModifyDevice",
  "module": "DeviceManager",
  "payload": {
    "modify": {
      "Ip": "0.0.0.0"
    },
    "uuid": "00000000-0000-0000-e2b0-34ddc8816c48"
  }
}
```

Modify ping360 IP adress

Needs:
- [x] https://github.com/bluerobotics/ping-viewer-next/pull/16

### CODE REVIEW

File: src/device/manager/mod.rs [hunk 1]
```diff
@@ -237,6 +249,12 @@ impl DeviceManager {
                     error!("DeviceManager: Failed to return GetDeviceHandler response: {e:?}");
                 }
             }
+            Request::ModifyDevice(request) => {
+                let answer = self.modify_device(request).await;
+                if let Err(e) = actor_request.respond_to.send(answer) {
+                    error!("DeviceManager: Failed to return ModifyDevice response: {e:?}");
+                }
```
Comment (joaoantoniocardoso) [inline src/device/manager/mod.rs:256]: you are using `err` in the rest of the code:

```suggestion
                if let Err(err) = actor_request.respond_to.send(answer) {
                    error!("DeviceManager: Failed to return ModifyDevice response: {err:?}");
                }
```

File: src/device/manager/mod.rs [hunk 2]
```diff
@@ -519,6 +537,49 @@ impl DeviceManager {
...
+                                return self.delete(request.uuid).await;
+                            }
+                            _ => {
+                                return Err(ManagerError::Other(format!(
+                                    "modify_device : invalid request for device : {request:?}"
+                                )))
+                            }
+                        }
+                    }
+                }
```
Comment (joaoantoniocardoso) [inline src/device/manager/mod.rs:559]:
```suggestion
                let device_info = self.info(request.uuid).await?;
                let Answer::DeviceInfo(data) = device_info else {
                    return Err(ManagerError::NoDevices);
                };
                
                let Some(info) = data.first() else {
                    return Err(ManagerError::NoDevices)
                };
                
                match &info.source {
                    SourceSelection::UdpStream(inner) => {
                        self.modify_device_ip(ip, inner.ip).await?;
                        return self.delete(request.uuid).await;
                    }
                    _ => {
                        return Err(ManagerError::Other(format!(
                            "modify_device : invalid request for device : {request:?}"
                        )))
                    }
                }
                
```

**Review by joaoantoniocardoso — CHANGES_REQUESTED** [review]
Some minor things

**Comment by RaulTrombin** [comment]
> Some minor things

Thks!

**Review by joaoantoniocardoso — APPROVED** [review]

**Pull request merged**